### ISSUE TITLE
Post Launch: Add a "source" field to CTMS contact

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-it/ctms-api#255]
With SFDC/SFMC, we previously had a "source" field on the contact itself, indicating the source of their very first email subscription. Currently in CTMS, we only track source on a per-newsletter basis. The Foundation needs to know the contact's original source because it tells them how this contact entered the funnel. Potential tasks here include:

* Add a column to the email table in CTMS and update CTMS logic to populate it moving forward
* Back-populate this field in CTMS using historical data from BQ
* Update the API to include this information explicitly 
* Update Cinchy to get this data from CTMS

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/CTMS-11)

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

The work involves: Brian --> Bryan --> Chris

Dependency for Marketing team in addition to MoFo

This work potentially needs to be split up into additional tasks after discussion from BBC

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

* This data is needed in SFNP(Target email campaigns), Acoustic & Cinchy
  * Make explicit field in CMTS (Chris to discuss with other devs)

Potential Impact: 

  * The longer we wait, there is a data gap.
  * Steph verified, not needed for newsletter source for upcoming fundraising campaign

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

This sprint: change to add columns to CTMS and API

Next:

  * Bryan to update to Acoustic
  * Chris - Backfill of historical data and hand over to Brian

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

Issue split into:

OSS442|Add a "source"field to CTMS  Update AcousticOSS443|Add a "source"field to CTMS  Backfill of historical data

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

This sprint add source column, engineers to meet to discuss this before starting on OSS-442 & OSS-443

**Comment by data-sync-user** [comment]
➤ Sarah Chu commented:

Brian Stack Did we get close with this item?

**Comment by grahamalama** [comment]
TODO: determine if this is still something that's wanted.
